src/test/scala/IntegrationTest.scala

```diff
@@ -261,6 +295,47 @@ class IntegrationTest extends WordSpec with Matchers with BeforeAndAfterAll with
     results.find(_.entity.id == personTwo.id).get.entity shouldBe personTwo
   }
 
+  "Should be able to do a batch update by keys" in {
+    val personOne = TestEntity(name = "Grover", age = 4, sex = Some("male"))
+    val otherPersonTwo = OtherTestEntity(name = "Other", location = "Berkeley")
+    val f1 = couchbase.insertDocument[TestEntity](personOne, personOne.id)
+    val f2 = couchbase.insertDocument[OtherTestEntity](otherPersonTwo, otherPersonTwo.id)
+    val insertResults = Await.ready(f1 zip f2, 10 seconds)
+
+    var personOneCas: Long = 0
+    var personTwoCas: Long = 0
+    insertResults.map{i =>
```

Comment:
Great place for partial function decomposition:
insertResults.map{ case (p1, p2) => (p1.cas, p2.cas)}
or something similar. It's always better than using x._1 x._2 IMO.